src/repositories/domains.ts

```diff
@@ -19,14 +19,29 @@ export class DomainsDb implements IDomainsDb {
     this.log = new Logger('DomainsDb')
   }
 
+  /**
+   * @description Get count of all domains from DB
+   * @returns {number}
+   */
+  async getCount (): Promise<number> {
+    const result = await this.db.query('SELECT count(*) OVER() AS total_count FROM domains', [])
+    let count = 0
+    if (result != null) {
+      count = Number(result?.rows[0]?.total_count)
+    }
+    return count
+  }
+
   /**
    * @description Get all Domains from DB
-   * @returns {AMTDomain[]} returns an array of AMT Domain objects
+   * @param {number} top
+   * @param {number} skip
+   * @returns {AMTDomain[]} returns an array of AMT Domain objects from DB
    */
-  async getAllDomains (): Promise<AMTDomain[]> {
-    const results = await this.db.query('SELECT name as Name, domain_suffix as DomainSuffix, provisioning_cert as ProvisioningCert, provisioning_cert_storage_format as ProvisioningCertStorageFormat, provisioning_cert_key as ProvisioningCertPassword FROM domains')
-    return await Promise.all(results.rows.map(async p => {
-      const result = mapToDomain(p)
+  async getAllDomains (top: number, skip: number): Promise<AMTDomain[]> {
```

Comment:
i think we should provide defaults here, `top:number = DEFAULT_SIZE, skip:number =0`

Reply:
done